app/recurring_job.go

```diff
@@ -750,7 +750,15 @@ func (job *Job) findARandomReadyNode(v *longhornclient.Volume) (string, error) {
 
 	var readyNodeList []string
 	for _, node := range nodeCollection.Data {
-		if readyConditionInterface, ok := node.Conditions[longhorn.NodeConditionTypeReady]; ok {
+		var readyConditionInterface map[string]interface{}
+		for i := range node.Conditions {
+			con := node.Conditions[i].(map[string]interface{})
+			if con["type"] == longhorn.NodeConditionTypeReady {
+				readyConditionInterface = con
+			}
+		}
```

Comment:
Consider use the function `GetCondition`.

Reply:
Sure.

Reply:
@jenting The data type of the condition is map[string]interface and not compatible with GetCondition's argument.
Do you have any idea for this?

Reply:
No, since it's the data struct for RESTful API. And as we discussed, we don't want to change the RESTful API interface.
So, let's keep it as is.